sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/OperatorStateMetadata.scala

```diff
@@ -166,18 +169,36 @@ object OperatorStateMetadataUtils extends Logging {
           s"version=${operatorStateMetadata.version}")
     }
   }
+
+  def getLastOffsetBatch(session: SparkSession, checkpointLocation: String): Long = {
```

Comment:
Nits: If we are calling this function only from test suites, shall we move it into test utils?

Reply:
We have similar logic in a couple of places already I believe. Prefer to keep this here to avoid multiple copies since it might be used by other callers in the future too